bin/lib/go_library_builder.py

```diff
@@ -0,0 +1,34 @@
+from typing import Dict, Any, List
+from lib.installation_context import InstallationContext
+from lib.library_build_config import LibraryBuildConfig
+
+class GoLibraryBuilder:
+    def __init__(
+        self,
+        logger,
+        language: str,
+        libname: str,
+        target_name: str,
+        install_context: InstallationContext,
+        buildconfig: LibraryBuildConfig,
+    ):
+        self.logger = logger
```

Comment:
the important part here is probably https://github.com/compiler-explorer/infra/blob/ea2996724b485453232dd68d502f0c8819cda8d8/bin/lib/library_builder.py#L122
which will give you information about the compilers that are available